lib/dest/write-contents/write-symbolic-link.js

```diff
@@ -1,18 +1,44 @@
 'use strict';
 
-var fs = require('graceful-fs');
+var path = require('path');
 
 var fo = require('../../file-operations');
 
 function writeSymbolicLink(file, optResolver, onWritten) {
-  // TODO handle symlinks properly
-  fs.symlink(file.symlink, file.path, function(symlinkErr) {
-    if (fo.isFatalOverwriteError(symlinkErr)) {
-      return onWritten(symlinkErr);
-    }
-
-    onWritten();
-  });
+  var srcPath = file.symlink;
+
+  var isDirectory = file.isDirectory();
```

Comment:
(Downloading the suggested patch so sending this now)

@erikkemperman The following example shows something a user might do that would require us to handle this case:

```js
vfs.src('./my-giant-image-dir')
  .pipe(through(function toSymlink(file, _, cb) {
    // Since a directory was sourced, file.isDirectory() will be true and we'll want to create a junction on Windows
    file.symlink = file.path;
  })
  .pipe(vfs.dest('./my-output-location'))
```